modules/tests/shared/src/test/scala/io/circe/CursorSuite.scala

```diff
@@ -0,0 +1,301 @@
+package io.circe
+
+import cats._
+import cats.syntax.all._
+import io.circe.syntax._
+import io.circe.tests.CirceMunitSuite
+import org.scalacheck.Prop
+import org.scalacheck.Prop._
+
+final class CursorSuite extends CirceMunitSuite {
+  import CursorSuite._
+
+  val j1: Json = Json.obj(
+    "a" -> (1 to 5).toList.asJson,
+    "b" -> Map("d" -> List(true, false, true)).asJson,
+    "c" -> Map("e" -> 100.1, "f" -> 200.2).asJson
+  )
+
+  val j2: Json = Json.obj(
+    "a" -> (0 to 5).toList.asJson,
+    "b" -> Map("d" -> List(true, false, true)).asJson,
+    "c" -> Map("e" -> 100.1, "f" -> 200.2).asJson
+  )
+
+  val j3: Json = Json.obj(
+    "a" -> (1 to 5).toList.asJson,
+    "b" -> 10.asJson,
+    "c" -> Map("e" -> 100.1, "f" -> 200.2).asJson
+  )
+
+  val j4: Json = Json.obj(
+    "a" -> (1 to 5).toList.asJson,
+    "c" -> Map("e" -> 100.1, "f" -> 200.2).asJson
+  )
+
+  val cursor: Cursor = Cursor.fromJson(j1)
+
+  property("focus should return the JSON value in a newly created cursor") {
+    forAll { (j: Json) =>
+      Cursor.fromJson(j).focus ?= Some(j)
+    }
+  }
+
+  property("Navigation into an object")(topProp)
+  private lazy val topProp = forAll { (j: Json) =>
```

Comment:
Why the `private lazy val` here?

Reply:
These are copied from the `ACursor` tests. The only changes I made were to make it compile with the new interface (which was always a simplification because `Cursor` is nicer to work with than `ACursor`).

I agree this is strange. I can update in this PR or a future one. I think the tests need a cleanup pass, I'm just not sure if you want that now or later.